File: 卡码/代码随想录/1/09-动态规划/C++/dp/05-子序列问题/src/03-编辑距离/LeetCode_0392.cpp
//392. 判断子序列
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	bool isSubsequence(string s, string t) {
		vector<vector<int>> dp(s.length()+1,vector<int>(t.length()+1,0));
		for(int i=1;i<s.length()+1;i++){
			for(int j=1;j<t.length()+1;j++){
				if(s[i-1]==t[j-1]) dp[i][j]=dp[i-1][j-1]+1;
				else dp[i][j]=dp[i][j-1];
			}
		}
		return dp[s.length()][t.length()]==s.length();
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/动态规划/_05_子序列问题/_03_编辑距离/LeetCode_0392.java
package com.kurumi.leetCode.codeThoughts.动态规划._05_子序列问题._03_编辑距离;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 392. 判断子序列
 */
public class LeetCode_0392 {
    static class Solution {
        public boolean isSubsequence(String s, String t) {
            int[][] dp = new int[s.length() + 1][t.length() + 1];
            for (int i = 1; i < s.length() + 1; i++) {
                for (int j = 1; j < t.length() + 1; j++) {
                    if (s.charAt(i - 1) == t.charAt(j - 1))
                        dp[i][j] = dp[i - 1][j - 1] + 1;
                    else
                        dp[i][j] = dp[i][j - 1];
                }
            }
            return dp[s.length()][t.length()] == s.length();
        }
    }
}
